Relax restrictions on the PPPoE src address, as per debian bug
293811:

  http://bugs.debian.org/cgi-bin/bugreport.cgi?bug=293811

Upstream-Status: Inappropriate [Backport from Debian]

--- rp-pppoe-3.8.orig/src/discovery.c
+++ rp-pppoe-3.8/src/discovery.c
@@ -376,8 +376,8 @@
 	if (!packetIsForMe(conn, &packet)) continue;
 
 	if (packet.code == CODE_PADO) {
-	    if (NOT_UNICAST(packet.ethHdr.h_source)) {
-		printErr("Ignoring PADO packet from non-unicast MAC address");
+	    if (BROADCAST(packet.ethHdr.h_source)) {
+		printErr("Ignoring broadcast PADO packet");
 		continue;
 	    }
 	    parsePacket(&packet, parsePADOTags, &pc);
